modules/RTC/TraceablePeerConnection.js

```diff
@@ -227,15 +318,133 @@ const dumpSDP = function(description) {
+ * @return {string}
+ */
+TraceablePeerConnection.prototype.getConnectionState = function() {
+    const state = this.peerconnection.iceConnectionState;
+
+    if (state === 'completed') {
+        return 'connected';
+    }
+
+    return state;
+};
+
+/**
+ * Tells whether or not this TPC instance is using Simulcast.
+ * @return {boolean} <tt>true</tt> if simulcast is enabled and active or
+ * <tt>false</tt> if it's turned off.
+ */
+TraceablePeerConnection.prototype.isSimulcastOn = function() {
+    return !this.options.disableSimulcast
+        && this.simulcast.isSupported()
+        && !this.isP2P;
+};
+
+/**
+ * Handles {@link SignalingEvents.PEER_VIDEO_TYPE_CHANGED}
+ * @param {string} endpointId the video owner's ID (MUC nickname)
+ * @param {VideoType} videoType the new value
+ * @private
+ */
+TraceablePeerConnection.prototype._peerVideoTypeChanged
+= function(endpointId, videoType) {
+    // Check if endpointId has a value to avoid action on random track
+    if (!endpointId) {
+        logger.error(`No endpointID on peerVideoTypeChanged ${this}`);
+
+        return;
+    }
+    const videoTrack = this.getRemoteTracks(endpointId, MediaType.VIDEO);
+
+    if (videoTrack.length) {
+        // NOTE 1 track per media type is assumed
+        videoTrack[0]._setVideoType(videoType);
+    }
+};
+
+/**
+ * Handles remote track mute / unmute events.
+ * @param {string} endpointId the track owner's identifier (MUC nickname)
+ * @param {MediaType} mediaType "audio" or "video"
+ * @param {boolean} isMuted the new mute state
+ * @private
+ */
+TraceablePeerConnection.prototype._peerMutedChanged
+= function(endpointId, mediaType, isMuted) {
+    // Check if endpointId is a value to avoid doing action on all remote tracks
+    if (!endpointId) {
+        logger.error('On peerMuteChanged - no endpoint ID');
+
+        return;
+    }
+    const track = this.getRemoteTracks(endpointId, mediaType);
+
+    if (track.length) {
+        // NOTE 1 track per media type is assumed
+        track[0].setMute(isMuted);
+    }
+};
+
+TraceablePeerConnection.prototype.getLocalTracks = function(mediaType) {
+    let tracks = getValues(this.localTracks);
+
+    if (mediaType !== undefined) {
+        tracks = tracks.filter(track => track.getType() === mediaType);
+    }
+
+    return tracks;
+};
+
+/**
+ * Obtains all remote tracks currently known to this PeerConnection instance.
+ * @param {string} [endpointId] the track owner's identifier (MUC nickname)
+ * @param {MediaType} [mediaType] the remote tracks will be filtered
+ * by their media type if this argument is specified.
+ * @return {Array<JitsiRemoteTrack>}
+ */
+TraceablePeerConnection.prototype.getRemoteTracks
+= function(endpointId, mediaType) {
+    const remoteTracks = [];
+    const endpoints
+        = endpointId ? [ endpointId ] : Object.keys(this.remoteTracks);
+
+    for (const endpoint of endpoints) {
+        const endpointTracks = this.remoteTracks[endpoint] || { };
```

Comment:
this goes away with Map conversion, but yes I believe there is space for both in the rules